crates/mev-share-rpc-api/src/types.rs

```diff
@@ -47,18 +53,92 @@ pub enum BundleItem {
         /// If true, the transaction can revert without the bundle being considered invalid.
         can_revert: bool,
     },
+    /// A bundle to backrun.
+    Bundle {
+        /// Params for the bundle.
+        bundle: SendBundleRequest,
+    },
+    /// By specifying an empty object in the body array, searchers can specify if, and where, 
+    /// other transactions may be included with relation to their bundle.
+    AllowBackrun { }
+}
+
+/// Requirements for the bundle to be included in the block.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ValidityPredicate {
+    /// Specifies the minimum percent of a given bundle's earnings to redistribute 
+    /// for it to be included in a builder's block.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub refund: Option<Vec<Refund>>,
+    /// Specifies what addresses should receive what percent of the overall refund for this bundle, 
+    /// if it is enveloped by another bundle (eg. a searcher backrun).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub refund_config: Option<Vec<RefundConfig>>,
 }
 
+/// Specifies the minimum percent of a given bundle's earnings to redistribute
+/// for it to be included in a builder's block.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Refund {
+    /// The index of the transaction in the bundle.
+    pub body_idx: u64,
+    /// The minimum percent of the bundle's earnings to redistribute.
+    pub percent: u64,
+}
+
+/// Specifies what addresses should receive what percent of the overall refund for this bundle,
+/// if it is enveloped by another bundle (eg. a searcher backrun).
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundConfig {
+    /// The address to refund.
+    pub address: Address,
+    /// The minimum percent of the bundle's earnings to redistribute.
+    pub percent: u64,
+}
+
+/// Preferences on what data should be shared about the bundle and its transactions
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PrivacyPredicate {
+    /// Hints on what data should be shared about the bundle and its transactions
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub hints: Option<Vec<PrivacyHint>>,
```

Comment:
while this is defined as an array, an object with boolean fields would be more useful here
so you can do `.privacy.hints.{has_calldata(), has_contract_address(),..}`

would require a wrapper type or object with custom serialized deserialize

Reply:
sounds good, gave this a try, PTAL when you get the chance